docs/rules/no-this-template-during-ssr.md

```diff
@@ -0,0 +1,34 @@
+# Disallow access of this.template during SSR (`lwc/no-this-template-during-ssr`)
+
+💼 This rule is enabled in the ✅ `recommended` config.
+
+<!-- end auto-generated rule header -->
+
+Browser APIs must not be accessed when SSR is being done. This rule prevents usage of browser APIs like `querySelector`
+on `template` in `connectedCallback` (and in methods called from `conenctedCallback` or anywhere when SSR is being done).
```

Comment:
We should recommend using `refs` hover `querySelector` for accessing rendered components. We should also add a ref code sample in the "Examples of **correct** code for this rule:".